### PR TITLE
Docs: help users connect to Prometheus using SigV4

### PR DESCRIPTION
**Pull request by mjseaman**
**What this PR does / why we need it**:
This PR adds some documentation for connecting to Amazon Managed Service for Prometheus from the Prometheus data source. We need it to help reduce friction that AMP users currently experience when trying to connect using non-AMG Grafana.

**Which issue(s) this PR fixes**:
No issues; this is related to a request from AWS on behalf of AMP users.

Fixes #

**Special notes for your reviewer**:
Reach out @mjseaman, @bergquist or @jongyllen for background

### CODE REVIEW

**Comment by mjseaman** [comment]
@rakyll FYI

**Comment by achatterjee-grafana** [comment]
This should be a separate topic in my opinion. We should not confuse folks using just Prometheus and Grafana. Also, keeping the information here will be a free advertisement for Amazon service.

@oddlittlebird and @daniellee, what do you think?

File: docs/sources/datasources/prometheus.md [hunk 1]
```diff
@@ -170,3 +170,9 @@ datasources:
     access: proxy
     url: http://localhost:9090
 ```
+
+## Amazon Managed Service for Prometheus
+
+The Prometheus data source works with Amazon Managed Service for Prometheus. If you are using an AWS Identity and Access Management (IAM) policy to control access to your Amazon Managed Service for Prometheus domain, then you must use AWS Signature Version 4 (AWS SigV4) to sign  all requests to that domain. For more details on AWS SigV4, refer to the [AWS documentation](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). Note that Grafana version 7.3.5 or higher is required to use SigV4 authentication.
```
Comment (oddlittlebird) [inline docs/sources/datasources/prometheus.md:176]:
```suggestion
The Prometheus data source works with Amazon Managed Service for Prometheus. If you are using an AWS Identity and Access Management (IAM) policy to control access to your Amazon Managed Service for Prometheus domain, then you must use AWS Signature Version 4 (AWS SigV4) to sign  all requests to that domain. For more details on AWS SigV4, refer to the [AWS documentation](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
```

File: docs/sources/datasources/prometheus.md [hunk 2]
```diff
@@ -170,3 +170,9 @@ datasources:
...
+
+To connect the Prometheus data source to Amazon Managed Service for Prometheus using SigV4 authentication, see the AWS guide to [Set up Grafana open source or Grafana Enterprise for use with AMP](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-standalone-grafana.html). If you are running Grafana in an Amazon EKS cluster, follow the AWS guide to [Query using Grafana running in an Amazon EKS cluster](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-grafana-7.3.html).
```
Comment (oddlittlebird) [inline docs/sources/datasources/prometheus.md:178]:
```suggestion
To connect the Prometheus data source to Amazon Managed Service for Prometheus using SigV4 authentication, refer to the AWS guide to [Set up Grafana open source or Grafana Enterprise for use with AMP](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-standalone-grafana.html). 

> **Note:** Grafana version 7.3.5 or higher is required to use SigV4 authentication.

If you are running Grafana in an Amazon EKS cluster, follow the AWS guide to [Query using Grafana running in an Amazon EKS cluster](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-grafana-7.3.html).
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

**Comment by oddlittlebird** [comment]
Looping in @scoren-gl for the Marketing perspective.

**Comment by mjseaman** [comment]
Worth mentioning that this is likely to be a temporary addition. Eventually we will want SigV4 auth to be consistent across AWS data source plugins (Cloudwatch, Timestream, X-Ray, IoT Sitewise, etc.) and once they are consistent I agree that SigV4 should be a separate topic linked from each of them. until then, the configuration is a little different from one plugin to the next.

**Comment by achatterjee-grafana** [comment]
Thanks for the clarification @mjseaman!

**Review by oddlittlebird — APPROVED** [review]
LGTM. Thanks for adding this, Mitch!

**Review by rakyll — APPROVED** [review]

**Pull request merged**